core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -388,7 +388,7 @@ class Partition(val topic: String,
     val leaderHWIncremented = inWriteLock(leaderIsrUpdateLock) {
       leaderReplicaIfLocal match {
         case Some(leaderReplica) =>
-          val outOfSyncReplicas = getOutOfSyncReplicas(leaderReplica, replicaMaxLagTimeMs)
+          val outOfSyncReplicas = getOutOfSyncReplicas(leaderReplica, replicaMaxLagTimeMs).filter(r => !r.ignoreIsrCheck)
```

Comment:
Good point. I am wondering if a simpler fix is to pass in lastCaughtUpTimeMs when constructing the Replica object and we can initialize it to current time.

BTW, do you see this issue on 0.10.1? In 0.10.1, lastCaughtUpTimeMsUnderlying actually initializes to current time in Replica.